java_src/foam/dao/DatastoreDAO.java

```diff
@@ -132,7 +133,7 @@ protected FObject deserialize(PropertyContainer e) {
         return obj;
     }
 
-    public Sink select(Sink sink, Integer skip, Integer limit, Comparator order, Predicate predicate) {
+    public Sink select_(X x, Sink sink, Integer skip, Integer limit, Comparator order, Predicate predicate) {
```

Comment:
Integer -> int

Integers create objects on the heap and so are more expensive to build and need to be GC'ed.